src/block/Hopper.php

```diff
@@ -92,7 +96,69 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 	}
 
 	public function onScheduledUpdate() : void{
-		//TODO
+		$world = $this->position->getWorld();
+
+		$facingBlock = $this->getSide($this->facing);
+		$topBlock = $this->getSide(Facing::UP);
+
+		$pushSuccess = false;
+		$pullSuccess = false;
+
+		if($topBlock instanceof HopperInteractable){
```

Comment:
The hopper starts to try to push items and next they pull.

Reply:
It's not resolved. The hopper will always try to retrieve the items above before pushing the items. It should be the other way around